alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/binder/AlfrescoStatusMetrics.java

```diff
@@ -0,0 +1,63 @@
+package eu.xenit.alfred.telemetry.binder;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import org.alfresco.repo.security.authentication.AuthenticationUtil;
+import org.alfresco.repo.transaction.RetryingTransactionHelper;
+import org.alfresco.service.cmr.admin.RepoAdminService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+
+public class AlfrescoStatusMetrics implements MeterBinder {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AlfrescoStatusMetrics.class);
+    private static final String STATUS_PREFIX = "alfresco.status";
+
+    private RepoAdminService repoAdminService;
+    private RetryingTransactionHelper retryingTransactionHelper;
+
+    public AlfrescoStatusMetrics(RepoAdminService repoAdminService, RetryingTransactionHelper retryingTransactionHelper) {
+        this.repoAdminService = repoAdminService;
+        this.retryingTransactionHelper = retryingTransactionHelper;
+    }
+
+    @Override
+    public void bindTo(@Nonnull MeterRegistry meterRegistry) {
+        LOGGER.info("Registering Alfresco Status metrics");
+        Gauge.builder(String.format("%s.%s", STATUS_PREFIX, "readonly"),
+                repoAdminService,
+                x -> getReadOnly(repoAdminService))
+                .description("Metric about Alfresco being in read-only mode")
+                .register(meterRegistry);
+
+    }
+
+    private double getReadOnly(RepoAdminService repoAdminService) {
```

Comment:
This can be a lot shorter, by using lambdas. 
I don't know if that's personal preference or if our code guide actually enforces that (my) preference. 

Also IIRC it's better to use `AuthenticationUtil.runAsSystem(...)`.

Reply:
I corrected to use AuthenticationUtil.runAsSystem

Reply:
So you don't like using lambda's?

Reply:
For lambda’s I have no strong feeling either way, I believe it’s more efficient if a hard-core developer spends 1 day to go through the whole code and improves code quality.
Perhaps you should start using gitlab before github, final product would look better and there would be more control.